src/Graphics/Canvas.purs

```diff
@@ -482,6 +483,15 @@ foreign import drawImageFull :: forall eff. Context2D -> CanvasImageSource -> Nu
 -- | Enumerates the different types of pattern repetitions.
 data PatternRepeat = Repeat | RepeatX | RepeatY | NoRepeat
 
+instance showPatternRepeat :: Show PatternRepeat where
+  show Repeat = "repeat"
```

Comment:
What I was getting at before was that we expect valid `Show` instances to produce strings which are executable code. So this and the `Show TextAlign` instance should be changed.

I was suggesting the current `show` be turned into `toString` which would be put into a `where` clause somewhere and not exported.